modules/migration-configuring-cors-3.adoc

```diff
@@ -0,0 +1,78 @@
+// Module included in the following assemblies:
+//
+// migration/migrating_openshift_3_to_4/migrating-openshift-3-to-4.adoc
+[id='migration-configuring-cors-3_{context}']
+= Configuring cross-origin resource sharing on an {product-title} 3 cluster
+
+You must configure cross-origin resource sharing on the {product-title} 3 cluster to enable communication between the CAM tool on the target cluster and the source cluster's API server.
+
+.Procedure
+
+. Log in to the {product-title} 4 cluster.
+. Obtain your CAM tool URL:
++
+----
+$ oc get -n openshift-migration route/migration -o go-template='(?i}//{{ .spec.host }}(:|\z){{ println }}' | sed 's,\.,\\.,g'
+----
+
+. Log in to the {product-title} 3 cluster.
+. Add the CAM tool URL to the `corsAllowedOrigins` stanza in the `/etc/origin/master/master-config.yaml` configuration file:
++
+----
+corsAllowedOrigins:
+- (?i)//migration-openshift-migration\.apps\.cluster\.com(:|\z) <1>
+- (?i)//openshift\.default\.svc(:|\z)
+- (?i)//kubernetes\.default(:|\z)
+----
+<1> Update the CAM tool URL.
++
+[NOTE]
+====
+This example uses the following syntax:
+
+* The `(?i)` makes it case-insensitive.
+* The `//` pins to the beginning of the domain and matches the double slash
+following `http:` or `https:`.
+* The `\.` escapes dots in the domain name.
+* The `(:|\z)` matches the end of the domain name `(\z)` or a port separator
+`(:)`.
+====
+
+. Restart the API server and controller manager to apply the changes:
++
+* In {product-title} 3.7 and 3.9, these components run as stand-alone host processes managed by `systemd` and are restarted by running the following command:
++
+----
+$ systemctl restart atomic-openshift-master-api atomic-openshift-master-controllers
+----
+
+* In {product-title} 3.10 and 3.11, these components run in static Pods managed by a kubelet and are restarted by running the following commands:
++
+----
+$ /usr/local/bin/master-restart api
+$ /usr/local/bin/master-restart controller
+----
+
+. Verify the configuration:
++
+----
+$ curl -v -k -X OPTIONS \
+"<cluster_url>/apis/migration.openshift.io/v1alpha1/namespaces/openshift-migration/migclusters" \ <1>
+-H "Access-Control-Request-Method: GET" \
+-H "Access-Control-Request-Headers: authorization" \
+-H "Origin: <CAM_tool_url>" <2>
```

Comment:
-H "Origin: https://<CAM_tool_url>"

Marco and myself lost time today trying to figure out why verification with not working.
Turns out we forgot to include 'http://' or 'https://' in the Origin when we attempted to verify.

The scheme is required based on how we defined the rule in master-config.yml.